src/6model/reprs/VMArray.c

```diff
@@ -812,8 +812,7 @@ static void copy_elements(MVMThreadContext *tc, MVMObject *src, MVMObject *dest,
     MVMArrayBody     *d_body      = (MVMArrayBody *)OBJECT_BODY(dest);
     MVMArrayREPRData *s_repr_data = REPR(src)->ID == MVM_REPR_ID_VMArray
                                     ? (MVMArrayREPRData *)STABLE(src)->REPR_data  : NULL;
-    MVMArrayREPRData *d_repr_data = REPR(src)->ID == MVM_REPR_ID_VMArray
-                                    ? (MVMArrayREPRData *)STABLE(dest)->REPR_data : NULL;
+    MVMArrayREPRData *d_repr_data = (MVMArrayREPRData *)STABLE(dest)->REPR_data;
```

Comment:
If `d_repr_data` can't ever be NULL, I'd say it should be removed from the if condition on line 8211/822.